asio/include/asio/experimental/impl/promise.hpp

```diff
@@ -64,10 +64,10 @@ struct promise_impl<void(Ts...), Executor, Allocator>
   Executor executor;
 
   template<typename Func, std::size_t... Idx>
-  void apply_impl(Func f, asio::detail::index_sequence<Idx...>)
+  void apply_impl(Func&& f, asio::detail::index_sequence<Idx...>)
```

Comment:
Here a const ref can do the trick - 'func' remains in scope and not a subject for a change.

Reply:
Cannot call `operator(...) &&` on a const ref...